src/pages/TeachersUnite/IntroSchoolPrincipalPage.js

```diff
@@ -74,6 +75,9 @@ function IntroSchoolPrincipalPage(props) {
             if (!_.isEmpty(values.partnerUserID) && !Str.isValidEmail(values.partnerUserID)) {
                 ErrorUtils.addErrorMessage(errors, 'partnerUserID', translate('teachersUnitePage.error.enterValidEmail'));
             }
+            if (!_.isEmpty(values.partnerUserID) && LoginUtils.isEmailPublicDomain(values.partnerUserID)) {
+                ErrorUtils.addErrorMessage(errors, 'partnerUserID', translate('teachersUnitePage.error.tryDifferentEmail'));
```

Comment:
NAB: Do you think we should use a more descriptive error message? Maybe something on the lines of "Please use an email address associated with your school"?

Reply:
Ah I think this message will be enough because we're already asking in the placeholder to enter a work email for the principal. This is only to cover the case when someone is trying to create a workspace with a public email, which probably is not an authentic school email.